frontend/src/app/components/NewsletterForm.js

```diff
@@ -75,36 +76,44 @@ export default class NewsletterForm extends React.Component {
       <label className={this.makeRevealedClassNames()} htmlFor={fieldName}>
         <input name={fieldName} type="checkbox" checked={this.props.privacy} required
                onChange={this.handlePrivacyClick} onClick={this.handlePrivacyClick} />
-        { this.state.privacyNote ? <span data-l10n-id="newsletterFormPrivacyAgreementRequired" style={{ color: 'red', marginRight: '0.5em' }}></span> : null }
-        <span data-l10n-id="newsletterFormPrivacyNotice">
-          I&apos;m okay with Mozilla handling by info as explained in
-          <a target="_blank" rel="noopener noreferrer"
-            href={url}>
-            this Privacy Notice
-          </a>.
-        </span>
+        { this.state.privacyNote ? <Localized id="newsletterFormPrivacyAgreementRequired">
+          <span style={{ color: 'red', marginRight: '0.5em' }}></span>
+        </Localized> : null }
+        <Localized id="newsletterFormPrivacyNotice">
+          <span>
+            I&apos;m okay with Mozilla handling by info as explained in
+            <a target="_blank" rel="noopener noreferrer"
```

Comment:
It's currently not possible to have nested elements inside of `Localized` components. Instead you may try passing the `<a>` element as a `$`-prefixed argument to `Localized`. See the bottom of https://github.com/projectfluent/fluent.js/wiki/Localized for an example of this approach.

Adding support for nested elements is definitely planned, although I haven't made much research into it and don't know how hard it's going to be.